common/src/main/java/com/scottlogic/deg/common/profile/constraints/atomic/NotConstraint.java

```diff
@@ -10,7 +10,7 @@
 public class NotConstraint implements AtomicConstraint {
     public final AtomicConstraint negatedConstraint;
 
-    protected NotConstraint(AtomicConstraint negatedConstraint) {
+    public NotConstraint(AtomicConstraint negatedConstraint) {
```

Comment:
Can you make this protected again, and call .negate on the constraints we want negating.